src/test/ui/impl-trait/issues/issue-78722.rs

```diff
@@ -8,6 +8,7 @@ struct Bug {
     V1: [(); {
         fn concrete_use() -> F {
             async {}
+            //~^ ERROR type mismatch resolving
```

Comment:
lol, why is this error showing up all of the sudden?

Reply:
Maybe it's the option being overwritten issue? Const qualifs are done before borrowck, so if borrowck succeeds but qualiffails, we never know

Reply:
> Maybe it's the option being overwritten issue?

Like overwriting `Some(_)` with `None` by accident? I don't think so.

I actually think it might be due to the fact that, since we put the tainted field into the MIR body itself, we have to load the MIR body before checking its tainted field. This might cause us to do some more computations on error paths than we did before. If so, then it's for that same reason I had to [add this logic](https://github.com/rust-lang/rust/pull/93691/files#diff-ec8e8ba5fe5c35f6cf8a40478ee1e03c46188228b9fb7b1d927c186d6f97021bR248-R253) to prevent a regression in the test `src/test/ui/issues/issue-77919.rs`.

I don't think this test emitting this extra error is necessarily a bad thing, though?

Reply:
no it's not a bad thing indeed.